%YAML:1.0

# Acceleration of gravity in the world frame.
# For example, in a RDF world, n_gravity = [0, 9.81, 0].
# NOTE: This is also used to determine the DEPTH axis (same as gravity).
n_gravity: [0, 9.81, 0]


imu0:
  frame_id: imu0

  # IMU and body frame are one and the same.
  body_T_imu:
    cols: 4
    rows: 4
    data: [1.0, 0.0, 0.0, 0.0,
           0.0, 1.0, 0.0, 0.0,
           0.0, 0.0, 1.0, 0.0,
           0.0, 0.0, 0.0, 1.0]
  noise_model:
    accel_noise_sigma:    0.0003924
    gyro_noise_sigma:     0.000205689024915
    accel_bias_rw_sigma:  0.004905
    gyro_bias_rw_sigma:   0.000001454441043

# NOTE(milo): Placeholder for acoustic positioning system.
aps0:
  frame_id: receiver0
  body_T_receiver:
    cols: 4
    rows: 4
    data: [1.0, 0.0, 0.0, 0.0,
           0.0, 1.0, 0.0, 0.0,
           0.0, 0.0, 1.0, -0.5,
           0.0, 0.0, 0.0, 1.0]

# FARMSIM CAMERA
# stereo0:
#   camera_left:
#     frame_id: camera_left
#     body_T_cam:
#       cols: 4
#       rows: 4
#       data: [1.0, 0.0, 0.0, -0.1,
#             0.0, 1.0, 0.0, 0.0,
#             0.0, 0.0, 1.0, 0.1,
#             0.0, 0.0, 0.0, 1.0]

#     # Camera specific definitions.
#     image_height: 480
#     image_width: 752
#     camera_model: pinhole
#     intrinsics: [415.876509, 415.876509, 375.5, 239.5] # fx, fy, cx, cy
#     distortion_model: radial-tangential
#     distortion_coefficients: [0.0, 0.0, 0.0, 0.0] # Zero means already undistorted.

#   camera_right:
#     frame_id: camera_right
#     body_T_cam:
#       cols: 4
#       rows: 4
#       data: [1.0, 0.0, 0.0, 0.1,
#             0.0, 1.0, 0.0, 0.0,
#             0.0, 0.0, 1.0, 0.1,
#             0.0, 0.0, 0.0, 1.0]

#     # Camera specific definitions.
#     image_height: 480
#     image_width: 752
#     camera_model: pinhole
#     intrinsics: [415.876509, 415.876509, 375.5, 239.5] # fx, fy, cx, cy
#     distortion_model: radial-tangential
#     distortion_coefficients: [0.0, 0.0, 0.0, 0.0] # Zero means already undistorted.

# # ZED MINI CAMERA (720 RESOLUTION)
# stereo0:
#   camera_left:
#     frame_id: camera_left
#     body_T_cam:
#       cols: 4
#       rows: 4
#       data: [1.0, 0.0, 0.0, 0.0,
#              0.0, 1.0, 0.0, 0.0,
#              0.0, 0.0, 1.0, 0.0,
#              0.0, 0.0, 0.0, 1.0]

#     # Camera specific definitions.
#     image_height: 720
#     image_width: 1280
#     camera_model: pinhole
#     intrinsics: [677.386963, 677.386963, 603.359863, 342.296387] # fx, fy, cx, cy
#     distortion_model: radial-tangential
#     distortion_coefficients: [0.0, 0.0, 0.0, 0.0] # Zero means already undistorted.

#   camera_right:
#     frame_id: camera_right
#     body_T_cam:
#       cols: 4
#       rows: 4
#       data: [1.0, 0.0, 0.0, 0.062939,
#              0.0, 1.0, 0.0, 0.0,
#              0.0, 0.0, 1.0, 0.0,
#              0.0, 0.0, 0.0, 1.0]

#     # Camera specific definitions.
#     image_height: 720
#     image_width: 1280
#     camera_model: pinhole
#     intrinsics: [677.386963, 677.386963, 603.359863, 342.296387] # fx, fy, cx, cy
#     distortion_model: radial-tangential
#     distortion_coefficients: [0.0, 0.0, 0.0, 0.0] # Zero means already undistorted.

# # ZED MINI CAMERA (VGA RESOLUTION)
# stereo0:
#   camera_left:
#     frame_id: camera_left
#     body_T_cam:
#       cols: 4
#       rows: 4
#       data: [1.0, 0.0, 0.0, 0.0,
#              0.0, 1.0, 0.0, 0.0,
#              0.0, 0.0, 1.0, 0.0,
#              0.0, 0.0, 0.0, 1.0]

#     # Camera specific definitions.
#     image_height: 376
#     image_width: 672
#     camera_model: pinhole
#     intrinsics: [336.135986, 336.135986, 317.032654, 178.710770] # fx, fy, cx, cy
#     distortion_model: radial-tangential
#     distortion_coefficients: [0.0, 0.0, 0.0, 0.0] # Zero means already undistorted.

#   camera_right:
#     frame_id: camera_right
#     body_T_cam:
#       cols: 4
#       rows: 4
#       data: [1.0, 0.0, 0.0, 0.062939,
#              0.0, 1.0, 0.0, 0.0,
#              0.0, 0.0, 1.0, 0.0,
#              0.0, 0.0, 0.0, 1.0]

#     # Camera specific definitions.
#     image_height: 376
#     image_width: 672
#     camera_model: pinhole
#     intrinsics: [336.135986, 336.135986, 317.032654, 178.710770] # fx, fy, cx, cy
#     distortion_model: radial-tangential
#     distortion_coefficients: [0.0, 0.0, 0.0, 0.0] # Zero means already undistorted.
